src/Servers/Kestrel/test/InMemory.FunctionalTests/BadHttpRequestTests.cs

```diff
@@ -47,6 +59,28 @@ public Task TestInvalidHeaders(string rawHeaders, string expectedExceptionMessag
                 expectedExceptionMessage);
         }
 
+        [Theory]
+        [MemberData(nameof(InvalidRequestHeaderDataLineFeedTerminator))]
+        public async Task TestInvalidHeadersQuirkMode(string rawHeaders, string expectedExceptionMessage)
+        {
+            var task = TestBadRequest(
+                $"GET / HTTP/1.1\r\n{rawHeaders}",
+                "400 Bad Request",
+                expectedExceptionMessage);
+
+            // These should not fail in quirk mode
+            if (_enableLineFeedTerminator)
+            {
+                // Unit test should fail since it expects a bad request
+                await Assert.ThrowsAnyAsync<Exception>(async () => await task);
```

Comment:
We should verify we actually get a successful response instead of catching any exception when we update this to remove all the variations and add the handful of tests specifically for `EnableLineFeedTerminator`